python/GafferSceneUI/SceneInspector.py

```diff
@@ -1308,6 +1337,70 @@ def __labelButtonPress( self, label, event ) :
 		script = self.__target.scene.ancestor( Gaffer.ScriptNode )
 		GafferSceneUI.ContextAlgo.setSelectedPaths( script.context(), IECore.PathMatcher( [ label.getText() ] ) )
 
+class _ShaderParameterSection( LocationSection ) :
+
+	class __Inspector( Inspector ) :
+
+		def __init__( self, shaderAttribute, parameterName = None ) :
+
+			Inspector.__init__( self )
+
+			self.__shaderAttribute = shaderAttribute
+			self.__parameterName = parameterName
+
+		def name( self ) :
+
+			return self.__parameterName
+
+		def __call__( self, target ) :
+
+			parameters = self.__parameters( target )
+			if parameters is None :
+				return None
+
+			return parameters.get( self.__parameterName )
+
+		def children( self, target ) :
+
+			parameters = self.__parameters( target )
+			if parameters is None :
+				return []
+
+			return [ self.__class__( self.__shaderAttribute, parameterName = p ) for p in parameters.keys() ]
+
+		def __parameters( self, target ) :
+
+			if target.path is None :
+				return None
+
+			attributes = target.attributes()
+
+			shaders = attributes[self.__shaderAttribute]
+
+			if not shaders :
+				return None
+
+			shader = shaders[0]  # only the first one is supported
```

Comment:
This should be `shaders[-1]`. The terminal shader for the network is the last one.

Reply:
Addressed in cb50163